### DISCUSSION TITLE
How to work with nested useArrayFields in an object ?

### DISCUSSION THREAD

**Discussion by MarieFG** [react-hook-form/react-hook-form#2501] [Q&A]
Hey!
I need some help on this nice hook :)
I've defaultValues with a nested array which is inside an child object (subItem). When I duplicate the item, there is a difference between the item itself and the result gave by the "getValues" function (as you can see in the console, the "subItem" level disapeared).
So when I duplicate, I can only have the high level informations such as the name, but not the nested ones (field1).
Could you help me please ?
Thanks,
MSS

https://codesandbox.io/embed/react-hook-form-usefieldarray-nested-arrays-0mmyg?fontsize=14&hidenavigation=1&theme=darkhttps://codesandbox.io/embed/react-hook-form-usefieldarray-nested-arrays-0mmyg?fontsize=14&hidenavigation=1&theme=darkhttps://codesandbox.io/embed/react-hook-form-usefieldarray-nested-arrays-0mmyg?fontsize=14&hidenavigation=1&theme=dark

**Comment by bluebill1049** [comment]
have u seen this example: https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-x7btr?

**Reply by MarieFG** [reply]
Thanks @bluebill1049 for your answer.
I saw this example, but the pb is I don't have the same defaultValues architecture => I have an object 'subItem' which contains the nestedArray.

Indees, my defaultValues looks like :

 `test: [
    {
      name: "useFieldArray1",
      subItem: {
        nestedArray: [{ field1: "field1" }, { field1: "field2" }]
      }
    }
  ]`

instead of

 `test: [
    {
      name: "useFieldArray1",
      nestedArray: [{ field1: "field1", field2: "field2" }]
    },`

**Reply by MarieFG** [reply]
Here you can see the difference between defaultValues and the result of the "getValues" function.

![image](https://user-images.githubusercontent.com/57953254/89645221-a19a1300-d8b9-11ea-8d5c-2fe59e979409.png)

The sandbox : https://codesandbox.io/s/react-hook-form-getvalues-with-nested-array-fields-inside-an-object-yh1bs

Thanks

**Reply by anthonycaron** [reply]
Hi @bluebill1049, I'm also interested in this issue. In fact in both examples there is the same issue. When trying to `append` an initialized item with `nestedArray` already filled, we can't see the items from `nestedArray` on the screen. Or maybe there's another way to add them ?

You can try it here, I updated your sandbox to reflect what I'm trying to describe: https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-add-initialized-item-44o57?file=/src/fieldArray.js

You will see that `newValue` has a prefilled `nestedArray` property but we're not able to see it on screen.

**Reply by bluebill1049** [reply]
Looks like i miss the reply above, what's the steps to reproduce?

**Reply by anthonycaron** [reply]
@bluebill1049 Thanks for looking into this.
There's only one step to reproduce the issue here, you can just **click the "APPEND" button**.

What we would **expect** is to see a new element with its Nested field array filled with values "field1" and "field2".
Something like this :
<img width="525" alt="expected" src="https://user-images.githubusercontent.com/5999761/90500872-c5781700-e14b-11ea-9dbd-a8ef33dc04fe.png">

But instead, the **result** is that we have a new element but there's no nested field array attached to it.
It looks like this :
<img width="527" alt="result" src="https://user-images.githubusercontent.com/5999761/90500930-dcb70480-e14b-11ea-9dd9-5afabf720a2d.png">

Here lies the issue. Let me know if you need more details and thanks for your time.

**Reply by bluebill1049** [reply]
useFieldArray hold reference to its own field array, in order to update the nested one, you can hoist ur useFieldArray to the parent level or trigger it at useEffect.

**Reply by anthonycaron** [reply]
Ok thanks !
